apps/commuter_rail_boarding/test/date_helpers_test.exs

```diff
@@ -28,12 +28,80 @@ defmodule DateHelpersTest do
         assert ~D[2018-11-03] = service_date(utc_datetime)
       end
 
-      assert ~D[2018-11-04] = service_date(local_dt!(~N[2018-11-04T02:30:00]))
+      assert ~D[2018-11-03] = service_date(local_dt!(~N[2018-11-04T02:30:00]))
       assert ~D[2018-11-04] = service_date(local_dt!(~N[2018-11-04T03:30:00]))
     end
   end
 
   describe "seconds_until_next_service_date/1" do
+    test "handles non-DST" do
+      assert 3601 =
+               seconds_until_next_service_date(
+                 service_date(local_dt!(~N[2018-02-11T01:59:59])),
+                 local_dt!(~N[2018-02-11T01:59:59])
+               )
+
+      assert 84_600 =
+               seconds_until_next_service_date(
+                 service_date(local_dt!(~N[2018-02-11T03:30:00])),
+                 local_dt!(~N[2018-02-11T03:30:00])
+               )
+
+      assert 81_000 =
+               seconds_until_next_service_date(
+                 service_date(local_dt!(~N[2018-02-11T04:30:00])),
+                 local_dt!(~N[2018-02-11T04:30:00])
+               )
+    end
+
+    test "handles DST transitions" do
+      # spring forward
+      # This is 1 second before the change, 1 second makes sense
+      assert 1 =
+               seconds_until_next_service_date(
+                 service_date(local_dt!(~N[2018-03-11T01:59:59])),
+                 local_dt!(~N[2018-03-11T01:59:59])
+               )
+
+      # This is 30 minutes after the change, so 23.5 hours makes sense:
+      assert 84_600 =
+               seconds_until_next_service_date(
+                 service_date(local_dt!(~N[2018-03-11T03:30:00])),
+                 local_dt!(~N[2018-03-11T03:30:00])
+               )
+
+      # At this point we are already a half hour past when it would have changed otherwise, so 22.5 hours makes sense:
+      assert 81_000 =
+               seconds_until_next_service_date(
+                 service_date(local_dt!(~N[2018-03-11T04:30:00])),
+                 local_dt!(~N[2018-03-11T04:30:00])
+               )
+
+      # fall back
+      {:ambiguous, dt_one, dt_two} = local_dt(~N[2018-11-04T01:30:00])
+
+      for {seconds_until, local_dt} <- %{5400 => dt_one, 1800 => dt_two} do
+        {:ok, utc_datetime} = DateTime.shift_zone(local_dt, "Etc/UTC")
+
+        assert seconds_until =
+                 seconds_until_next_service_date(service_date(utc_datetime), utc_datetime)
+      end
```

Comment:
question: I'm a bit confused by this part. Why are we testing the ambiguous date times? Only one of them is the correct one.

Reply:
> question: I'm a bit confused by this part. Why are we testing the ambiguous date times? Only one of them is the correct one.

@bklebe This is copied logic from another test, but basically what it is doing is checking the seconds remaining are correct at both instances of 1:30am. So the seconds remaining will be different at both 1:30s (5400 before the timechange, 1800 the second time around) 🥴